source/fab/steps/grab/git.py

```diff
@@ -3,113 +3,98 @@
 #  For further details please refer to the file COPYRIGHT
 #  which you should have received as part of this distribution
 # ##############################################################################
+import warnings
+from abc import ABC
 from pathlib import Path
 from typing import Union, Dict
 
 from fab.steps.grab import GrabSourceBase
+from fab.tools import run_command
 
-try:
-    import git
-except ImportError:
-    git = None  # type: ignore
 
+def current_commit(folder=None):
+    folder = folder or '.'
+    output = run_command(['git', 'log', '--oneline', '-n', '1'], cwd=folder)
+    commit = output.split()[0]
+    return commit
 
-class GrabGit(GrabSourceBase):
-    """
-    Grab a Git repo into the project workspace.
-
-    A destination name must be specified because each git repo needs to be in a separate folder.
-
-    .. note::
-
-        A revision must be specified.
-        If `shallow` is set (the default), then the revision should be a branch or tag.
-        If `shallow` is not set, the revision can also be a commit hash.
-
-    A `shallow` grab clones/fetches only the given revision, with no history.
-    Otherwise, the full repo is cloned including all branches and history.
 
-    Example::
-
-        GrabGit(src='~/git/my_project', revision='v0.1b')
-        GrabGit(src='https://github.com/me/my_project.git', revision='a1b2c3', shallow=False)
+class GrabGitBase(GrabSourceBase, ABC):
+    """
+    Base class for Git operations.
 
     """
+    def run(self, artefact_store: Dict, config):
```

Comment:
No docstring

Reply:
That one is inherited, and is the same for all  subclasses of `GrabSourceBase`.
I rendered the docs to double check.

![git_doc](https://user-images.githubusercontent.com/2048530/223429177-14c088fd-bc1e-4ea5-b771-d002302d1286.png)

Reply:
Cool, thanks for checking.